pallets/parachain-staking/src/lib.rs

```diff
@@ -3376,6 +3401,20 @@ pub mod pallet {
 				(0u32, 0u32, BalanceOf::<T>::zero());
 			// choose the top TotalSelected qualified candidates, ordered by stake
 			let collators = Self::compute_top_candidates();
+			if collators.is_empty() {
+				// SELECTION FAILED TO SELECT >=1 COLLATOR => select collators from previous round
+				let last_round = now.saturating_sub(1u32);
```

Comment:
Does this need a special case for round 0?

Reply:
I don't think so. If none are set in genesis, then it would select nobody but there is no way around this without having the "trusted" collator backup set that you mentioned.

Nobody should launch without candidates set in genesis. It would be better to emit an error in the genesis config than to set it here.

Reply:
I see we dont update here `SelectedCandidates`. Is it because it remains identical to the previous?

Reply:
Also shall we maybe emit the CollatorChosen event? there might be tools that rely on this event been emitted, hence my question

Reply:
Ah sorry, just saw the comment `// `SelectedCandidates` remains unchanged from last round`

Reply:
> Also shall we maybe emit the CollatorChosen event? there might be tools that rely on this event been emitted, hence my question

Yes, I'm adding this now because that's a good assumption.